L1Trigger/L1THGCal/interface/HGCalTriggerGeometryBase.h

```diff
@@ -92,18 +89,18 @@ class HGCalTriggerGeometryBase {
   virtual unsigned triggerLayer(const unsigned id) const = 0;
 
 protected:
-  void setCaloGeometry(const edm::ESHandle<CaloGeometry>& geom) { calo_geometry_ = geom; }
-  void setEEGeometry(const edm::ESHandle<HGCalGeometry>& geom) { hgc_ee_geometry_ = geom; }
-  void setHSiGeometry(const edm::ESHandle<HGCalGeometry>& geom) { hgc_hsi_geometry_ = geom; }
-  void setHScGeometry(const edm::ESHandle<HGCalGeometry>& geom) { hgc_hsc_geometry_ = geom; }
+  void setCaloGeometry(const CaloGeometry* geom) { calo_geometry_ = geom; }
+  void setEEGeometry(const HGCalGeometry* geom) { hgc_ee_geometry_ = geom; }
+  void setHSiGeometry(const HGCalGeometry* geom) { hgc_hsi_geometry_ = geom; }
+  void setHScGeometry(const HGCalGeometry* geom) { hgc_hsc_geometry_ = geom; }
 
 private:
   const std::string name_;
 
-  edm::ESHandle<CaloGeometry> calo_geometry_;
-  edm::ESHandle<HGCalGeometry> hgc_ee_geometry_;
-  edm::ESHandle<HGCalGeometry> hgc_hsi_geometry_;
-  edm::ESHandle<HGCalGeometry> hgc_hsc_geometry_;
+  const CaloGeometry* calo_geometry_ = nullptr;
+  const HGCalGeometry* hgc_ee_geometry_ = nullptr;
+  const HGCalGeometry* hgc_hsi_geometry_ = nullptr;
+  const HGCalGeometry* hgc_hsc_geometry_ = nullptr;
```

Comment:
(repeating from the description) There is a slight change in behavior. Earlier, the existence of these products was checked by the `isValid()` or `product()` calls in above. This PR suggests to move those checks to the ESProducer. Is that ok?

Reply:
This looks fine to me (actually, probably slightly better than the previous approach)